### PR TITLE
Ex6

### PR DESCRIPTION
**Pull request by galtoubul**

### CODE REVIEW

File: src/ex4/client/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/build
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
```
Comment (liorshamian) [inline src/ex4/client/.gitignore]: you could simply add *.log*

File: src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect, useRef } from "react";
```
Comment (liorshamian) [inline src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx:1]: useMemo is missing

File: src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import PropTypes from "prop-types";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+
+library.add(faPlus);
+const validTask = "valid-task-btn";
+
+const isValidTask = (task) => {
+  return !/^\s*$/.test(task);
```
Comment (liorshamian) [inline src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx:12]: I was able to add many "invalid" tasks

**Review by liorshamian — APPROVED** [review]
why is everything under folder ex4?

when I ran npm start I got sh: nodemon: command not found

README file is missing (had to do many steps by myself with no documentation like installing nodemon, npx sequelize-cli db:migrate from specific folder, etc.)

but besides all that all in all good structure & good job

File: src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx (hunk 3 above)
Comment (galtoubul) [reply src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx:12]: I couldn't add "invalid" (space only) tasks. What were the actions you made?

https://user-images.githubusercontent.com/58177619/177714648-f8b9c264-30cd-429f-8f17-4a5b74bc9cd8.mp4

:

File: src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx (hunk 2 above)
Comment (galtoubul) [reply src/ex4/client/src/components/AddTaskInput/AddTaskInput.jsx:1]: Just to make sure, by "missing" do you mean that I should have use "useMemo" in AddTaskInput?
If so, what do you think I should memoize?

**Pull request merged**